### PR TITLE
adds hosting goal support to import/export table

### LINKED ISSUE #3850: Exporting and importing a table does not maintain hosting goal
When a table is exported and them imported, the hosting goal is not maintained.  Need to the add the hosting goal to the exported metadata.  On import, if the hosting goal is absent (importing a table from an older version) then could set it to the default.

This is causing new ComprehensiveIT test added in #3835 to fail.

### PR DESCRIPTION
**Pull request by keith-turner**
Modified export table to write out the hosting goal. Modified import table to read the hosting goal or set it to a default if not seen. Also added setting and getting hosting goals to ComprehensiveIT

fixes #3850

### CODE REVIEW

File: server/manager/src/main/java/org/apache/accumulo/manager/tableOps/tableImport/PopulateMetadataTable.java [hunk 1]
```diff
@@ -149,6 +153,11 @@ public Repo<Manager> call(long tid, Manager manager) throws Exception {
             if (m == null || !currentRow.equals(metadataRow)) {
 
               if (m != null) {
+                if (!sawHostingGoal) {
+                  // add a default hosting goal
+                  HostingColumnFamily.GOAL_COLUMN.put(m,
+                      new Value(TabletHostingGoal.ONDEMAND.name()));
```
Comment (dlmarion) [inline server/manager/src/main/java/org/apache/accumulo/manager/tableOps/tableImport/PopulateMetadataTable.java:159]:
```suggestion
                      TabletHostingGoalUtil.toValue(TabletHostingGoal.ONDEMAND);
```

Just a suggestion, feel free to ignore. This adds a method call, but would make this consistent with other places in the code. For reference, this is what `TabletMutatorBase.putHostingGoal` does.

**Review by dlmarion — APPROVED** [review]

File: server/manager/src/main/java/org/apache/accumulo/manager/tableOps/tableImport/PopulateMetadataTable.java [hunk 2]
```diff
@@ -161,11 +171,22 @@ public Repo<Manager> call(long tid, Manager manager) throws Exception {
...
+              sawHostingGoal = true;
+            }
             m.put(key.getColumnFamily(), cq, val);
 
             if (endRow == null && TabletColumnFamily.PREV_ROW_COLUMN.hasColumns(key)) {
+
+              if (!sawHostingGoal) {
+                // add a default hosting goal
+                HostingColumnFamily.GOAL_COLUMN.put(m,
+                    new Value(TabletHostingGoal.ONDEMAND.name()));
```
Comment (dlmarion) [inline server/manager/src/main/java/org/apache/accumulo/manager/tableOps/tableImport/PopulateMetadataTable.java:187]: There is another location here...

**Comment by EdColeman** [comment]
Why not have the hosting goal set as part of the import command?  Or at least support overriding the goal on import?

I guess it depends on usage and what ever is picked will be an inconvenience to someone else.  If I am exporting / importing to transfer files to set up a mirror destination, then having the goal state mirrored as part of the configuration would reduce the amount of "post configuration" needed.

On the other hand, if I was exporting / importing to a destination it may be because I have different usages in mind and just because it was configured one way on the source, I may want or need a different goal at the destination.

A third possibility may be to always import into an on-demand (or whatever state has the lowest resource demands) - that way it is a post configuration step to set the imported table as online if that is the desired goal.  That way, the import would have the least resource impact until specifically changed by the user.

**Comment by keith-turner** [comment]
> Why not have the hosting goal set as part of the import command? Or at least support overriding the goal on import?

That could be done, I think it would be best done as a follow on issue.  It seems we have  a new [ImportConfiguration](https://github.com/apache/accumulo/blob/4de6087f2739602128bbe22ebb9ee1c020810c7c/core/src/main/java/org/apache/accumulo/core/client/admin/ImportConfiguration.java#L31) interface that was added in 2.1 that would make adding this not disruptive.

Wondering about the following.  Wondering if it will work and if it should work.

 * Import table setting the keep offline setting
 * Set the hosting goals on the offline table (does this work, if so should it work?)
 * Bring the table online

**Comment by keith-turner** [comment]
Opened #3856 for setting hosting goals on offline table.  @EdColeman can you open an issue about setting hosting goals on table import?

**Pull request merged**